### ISSUE TITLE
OnStop return error

### ISSUE THREAD

**Issue by codeczx** [protyposis/ITEC-MediaPlayer#41]
Hi,
I called `pause()` in `surfaceDestroyed()` and `start()` in `surfaceCreated()`.
https://gist.github.com/codeczx/cd11b393f80f36b6a4dea581f4f1ad63

Then i pressed home button and return,but app crash and log show (not always,but exist)
``` 
E/MediaPlayer: decoder error, too many instances?
               android.media.MediaCodec$CodecException: Error 0xffffffde
                   at android.media.MediaCodec.releaseOutputBuffer(Native Method)
                   at android.media.MediaCodec.releaseOutputBuffer(MediaCodec.java:2424)
                   at net.protyposis.android.mediaplayer.MediaCodecVideoDecoder.releaseFrame(MediaCodecVideoDecoder.java:97)
                   at net.protyposis.android.mediaplayer.MediaCodecVideoDecoder.renderFrame(MediaCodecVideoDecoder.java:89)
                   at net.protyposis.android.mediaplayer.MediaPlayer$PlaybackThread.renderVideoFrame(MediaPlayer.java:1231)
                   at net.protyposis.android.mediaplayer.MediaPlayer$PlaybackThread.loopInternal(MediaPlayer.java:1055)
                   at net.protyposis.android.mediaplayer.MediaPlayer$PlaybackThread.handleMessage(MediaPlayer.java:906)
                   at android.os.Handler.dispatchMessage(Handler.java:98)
                   at android.os.Looper.loop(Looper.java:158)
                   at android.os.HandlerThread.run(HandlerThread.java:61)
E/AndroidRuntime: FATAL EXCEPTION: MediaPlayer#PlaybackThread
                  Process: com.zhixianshi.peony, PID: 3923
                  android.media.MediaCodec$CodecException: Error 0xffffffde
                      at android.media.MediaCodec.releaseOutputBuffer(Native Method)
                      at android.media.MediaCodec.releaseOutputBuffer(MediaCodec.java:2424)
                      at net.protyposis.android.mediaplayer.MediaCodecDecoder.releaseFrame(MediaCodecDecoder.java:432)
                      at net.protyposis.android.mediaplayer.MediaPlayer$PlaybackThread.releaseInternal(MediaPlayer.java:1177)
                      at net.protyposis.android.mediaplayer.MediaPlayer$PlaybackThread.handleMessage(MediaPlayer.java:936)
                      at android.os.Handler.dispatchMessage(Handler.java:98)
                      at android.os.Looper.loop(Looper.java:158)
                      at android.os.HandlerThread.run(HandlerThread.java:61)
```
I can call `release()` in `surfaceDestroyed` to solve this problem,but this kind of situation seem not necessary to release player and create again.

**Comment by protyposis** [comment]
Thanks for the sample code, this is fixed in v4.2.1. When returning back, you'll notice video coding artifacts until the next sync frame. That's because the codec needs to be recreated when setting a new surface, whereby reference frames get lost. If you don't want these artifacts, you'll have to `release()` and create a new instance, or issue a `seek()` to the current position after `setSurface`.

I recommend you to always `release()` the MediaPlayer when the Activity/Fragment goes into the background, otherwise it keeps holdings resources (e.g. hardware decoders) that other apps might need in the meantime. You can never know when or if the user returns back to your app.

See also what [Android's MediaPlayer documentation](https://developer.android.com/reference/android/media/MediaPlayer.html) says about this:

> It is also recommended that once a MediaPlayer object is no longer being used, call release() immediately so that resources used by the internal player engine associated with the MediaPlayer object can be released immediately. Resource may include singleton resources such as hardware acceleration components and failure to call release() may cause subsequent instances of MediaPlayer objects to fallback to software implementations or fail altogether. Once the MediaPlayer object is in the End state, it can no longer be used and there is no way to bring it back to any other state.

**Comment by codeczx** [comment]
Thanks.
